plugin-taskbar/lxqttaskgroup.cpp

```diff
@@ -657,18 +660,29 @@ bool LXQtTaskGroup::onWindowChanged(WId window, NET::Properties prop, NET::Prope
 
         bool set_urgency = false;
         bool urgency = false;
+
+        if (auto *x11Application = qGuiApp->nativeInterface<QNativeInterface::QX11Application>())
+        {
+            WId appRootWindow = XDefaultRootWindow(x11Application->display());
+            urgency = NETWinInfo(x11Application->connection(), window, appRootWindow, NET::Properties{}, NET::WM2Urgency).urgency();
+        }
+
         if (prop2.testFlag(NET::WM2Urgency))
         {
             set_urgency = true;
-            urgency = NETWinInfo(QX11Info::connection(), window, QX11Info::appRootWindow(), NET::Properties{}, NET::WM2Urgency).urgency();
         }
         if (prop.testFlag(NET::WMState))
         {
             KWindowInfo info{window, NET::WMState};
+
+            // Force refresh urgency
             if (!set_urgency)
-                urgency = NETWinInfo(QX11Info::connection(), window, QX11Info::appRootWindow(), NET::Properties{}, NET::WM2Urgency).urgency();
-            std::for_each(buttons.begin(), buttons.end(), std::bind(&LXQtTaskButton::setUrgencyHint, std::placeholders::_1, urgency || info.hasState(NET::DemandsAttention)));
-            set_urgency = false;
+            {
+                //TODO: maybe do it in common place with NET::WM2Urgency
+                std::for_each(buttons.begin(), buttons.end(), std::bind(&LXQtTaskButton::setUrgencyHint, std::placeholders::_1, urgency || info.hasState(NET::DemandsAttention)));
+                set_urgency = false;
+            }
+
```

Comment:
`set_urgency = false;` inside `if (!set_urgency)`?

Making such changes in a PR that isn't about them will cause complications.

@palinek 
Would you please look into this? The code is yours, I think.

Reply:
@gfgit 
`set_urgency = false` had a correct place before. You also changed the place where `urgency` was assigned a few lines above. Would you make the code as it was before (of course, except for using `nativeInterface`)?

Reply:
Original logic was badly written. Anyway this code will go away when moving to WM Abstraction, so we can keep it for now.

Reply:
I'm afraid, this isn't still the original code

> Original logic was badly written

Even if so, we should not make such changes in this PR; otherwise, it'll be very difficult to find causes of problems. For having a clean and searchable history, each job must be done in a separate PR.

Please reverse all changes here, except for using `nativeInterface`!

Reply:
You realize this is quite silly? Moving 2 lines in a common place is far more readable than squashing 10 unrelated commits together...

Reply:
> You realize this is quite silly?

On the contrary, now we know that this PR won't change anything in this regard after being merged. So, if some problem exists, we won't waste our time. That's how I found the cause of localization problem in Worldclock (https://github.com/lxqt/lxqt-panel/pull/2024#issuecomment-2009619729).